az-digital/az_quickstart
# Fixes #390, Fixes #427, Fixes #538, Fixes #613 Card Widget Updates

This pull request updates many aspects of the card widget of our custom card field, primarily the introduction of card previews and two-step editor forms so that the entire deck does not open at once.

## Description
The following changes are made to the card type and card widgets:

- Storage of the media ID is sanitized so that it may be NULL
- Corrected issue where the media widget was providing a comma delimited list of Media IDs
- Card field forms are now a two-step AJAX process where an `Edit Card` button is pressed to edit an individual card, as opposed to all field inputs showing simultaneously.
- Card background color selection
- Card background color implementation
- On the edit screen, a rendered preview is provided for collapsed cards.

## Related Issue
#538 
#427 
#390 
#613 
#377 

## How Has This Been Tested?

- Create a page
- Add cards
- Verify that for new cards they do not initially appear collapsed (eg. all fields are accessible for cards newly added to the entity)
- Save page
- Edit page
- Edit cards paragraph
- Verify that each card you added has a visible preview, but that the form fields are hidden
- Click `Edit Card` and verify it expands to make its form fields accessible.
- Make changes
- Verify that an image can be removed from the card (Fixes #538)
- Click `Collapse Card` and verify it reverts back to preview mode with changes reflected. (or skip this step and leave it expanded)
- Save page
- Verify changes were updated
- Verify background colors can be set for cards
- Verify background colors appear for cards.

## Types of changes


## Checklist:
